src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,99 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+import React, { useEffect, useMemo, useState } from 'react';
+// import debounce from 'lodash.debounce';
+
+type Props = {
+  onSelect: (value: boolean | null) => void;
+  onChange: (newQuery: string) => void;
+};
+
+export const TodoFilter: React.FC<Props> = ({ onSelect, onChange }) => {
+  // #region select
+  const [selectValue, setSelectValue] = useState('');
```

Comment:
it makes sense to pass `filter.completed` and store the correct string value based on it. Otherwise, rename `completed` to status and save string (in this case it's better to create a union type) there